src/IW/Sync/Search.hs

```diff
@@ -1,31 +1,60 @@
-module IW.Server.Search where
+module IW.Sync.Search
+       ( fetchAllHaskellRepos
+       , fetchAllHaskellIssues
+       , fetchOpenIssuesWithLabels
+       ) where
 
-import GitHub (Error, Repo, SearchResult, Issue)
+import GitHub (SearchResult (..))        
 import GitHub.Endpoints.Search (searchRepos, searchIssues)
 
+import IW.App (WithError)
+import IW.Db (WithDb)
+import IW.App.Error (AppErrorType, throwError)
+
+import qualified GitHub (Repo, Issue, Error (..))
+
+
 -- | Fetch all repositories with Haskell language
-fetchAllHaskellRepos :: IO (Either Error (SearchResult Repo))
+fetchAllHaskellRepos :: IO (Either GitHub.Error (SearchResult GitHub.Repo))
 fetchAllHaskellRepos = searchRepos "language:haskell"
 
+githubErrToAppErr :: GitHub.Error -> AppErrorType
+githubErrToAppErr = undefined
+
+syncAllHaskellRepos :: (MonadIO m, WithDb env m, WithError m) => m ()
+syncAllHaskellRepos = do
+    result <- liftIO fetchAllHaskellRepos
+    case result of
+        Right (SearchResult count vec) -> undefined 
+        Left err                       -> throwError . githubErrToAppErr $ err
+        
+{-
 -- | Fetch all repositories with Haskell language and label "help wanted"
 fetchHaskellReposHW :: IO (Either Error (SearchResult Repo))
 fetchHaskellReposHW = searchRepos "language:haskell help-wanted-issues:>0"
 
 -- | Fetch all repositories with Haskell language and label "good first issue"
 fetchHaskellReposGFI :: IO (Either Error (SearchResult Repo))
 fetchHaskellReposGFI = searchRepos "language:haskell good-first-issues:>0"
+-}
+
+-- | Fetch all open issues with Haskell language
+fetchAllHaskellIssues :: IO (Either GitHub.Error (SearchResult GitHub.Issue))
+fetchAllHaskellIssues = searchIssues "language:haskell is:open"
```

Comment:
Could you please create an issue (for future) to implement garbage collecting of old issues? This will require to fetch all closed issues and delete all closed issues from our database. We don't need to implement this now, but it's good to keep track of that issue for future.

Reply:
Good idea